core_impl/src/main/java/io/opencensus/stats/MutableAggregation.java

```diff
@@ -33,6 +39,20 @@ private MutableAggregation() {
   abstract void add(double value);
 
   /**
+   * Update the summary stats by removing an expired value.
+   *
+   * @param value the expired value to be removed from population.
+   */
+  abstract void remove(double value);
```

Comment:
Different window types will likely require different implementation approaches so splitting MutableAggregations along these lines probably makes the most sense. For interval stats, we can go with the straightforward approach and then optimize for different measures later (e.g. keeping a running sum).